scipy/sparse/dok.py

```diff
@@ -427,8 +427,8 @@ def transpose(self, axes=None, copy=False):
         M, N = self.shape
         new = dok_matrix((N, M), dtype=self.dtype, copy=copy)
 
-        for key, value in iteritems(self):
-            new[key[1], key[0]] = value
+        for (lhs, rhs), value in iteritems(self):
+            new[rhs, lhs] = value
```

Comment:
I wasn't sure about this either but since there is a swap between the members I figured better not to use ``i, j`` since they look similar but now looking this ``rhs, lhs`` does not look much better either.